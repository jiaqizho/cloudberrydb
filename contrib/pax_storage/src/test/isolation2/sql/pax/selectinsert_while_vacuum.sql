-- @Description Ensures that an insert during a vacuum operation is ok
-- 
CREATE TABLE selectinsert_while_vacuum_pax_tbl (a INT);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);

DELETE FROM selectinsert_while_vacuum_pax_tbl WHERE a < 128;
4: BEGIN;
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
5: BEGIN;
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
4: BEGIN;
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
2>: VACUUM selectinsert_while_vacuum_pax_tbl;
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;BEGIN;insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);COMMIT;
2<:
3: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl WHERE a = 1500;
3: INSERT INTO selectinsert_while_vacuum_pax_tbl VALUES (0);
