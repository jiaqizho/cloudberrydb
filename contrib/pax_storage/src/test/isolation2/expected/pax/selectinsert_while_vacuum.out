-- @Description Ensures that an insert during a vacuum operation is ok
--
CREATE TABLE selectinsert_while_vacuum_pax_tbl (a INT);
CREATE
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000
insert into selectinsert_while_vacuum_pax_tbl select generate_series(1,1000);
INSERT 1000

DELETE FROM selectinsert_while_vacuum_pax_tbl WHERE a < 128;
DELETE 2667
4: BEGIN;
BEGIN
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
 count 
-------
 18333 
(1 row)
5: BEGIN;
BEGIN
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
 count 
-------
 18333 
(1 row)
4: BEGIN;
BEGIN
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;
 count 
-------
 18333 
(1 row)
2>: VACUUM selectinsert_while_vacuum_pax_tbl;  <waiting ...>
4: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl;BEGIN;insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);insert into selectinsert_while_vacuum_pax_tbl select generate_series(1001,2000);COMMIT;
SELECT
2<:  <... completed>
VACUUM
3: SELECT COUNT(*) FROM selectinsert_while_vacuum_pax_tbl WHERE a = 1500;
 count 
-------
 20    
(1 row)
3: INSERT INTO selectinsert_while_vacuum_pax_tbl VALUES (0);
INSERT 1
